lib/cli/src/commands/publish.rs

```diff
@@ -27,6 +27,16 @@ pub struct Publish {
     /// Defaults to current working directory.
     #[clap(name = "PACKAGE_PATH")]
     pub package_path: Option<String>,
+    /// Wait for package to be available on the registry before exiting.
+    #[clap(long)]
+    pub wait: bool,
+    /// Timeout (in seconds) for the publish query to the registry.
+    ///
+    /// Note that this is not the timeout for the entire publish process, but
+    ///
+    /// for each individual query to the registry during the publish flow.
+    #[clap(long, default_value = "30")]
+    pub timeout: u64,
```

Comment:
We should probably use [`humantime::Duration`](https://docs.rs/humantime/latest/humantime/struct.Duration.html) instead of `u64` here. That'll let users do `--timeout="1 min 42s"` or whatever.

There is also a convenient `.into()` implementation for converting to a normal `std::time::Duration` that we can use when initializing the `Publish` struct.

```suggestion
    #[clap(long, default_value = "30s")]
    pub timeout: humantime::Duration,
```

(probably requires adding `humantime` to `wasmer-cli` as a dependency.

Reply:
@Michael-F-Bryan this is pretty cool, but I noticed that humantime is not a dependency for wasmer-cli. Are we ok with adding it as a new dependency?

Reply:
I added the dependency, closing this.